pkg/overlay/cache_test.go

```diff
@@ -296,19 +296,22 @@ func TestRedisPut(t *testing.T) {
 
 	for _, c := range putCases {
 		t.Run(c.testID, func(t *testing.T) {
-			db, cleanup := boltTestClient(t, c.data)
-			defer cleanup()
-
+			db := redisTestClient(t, c.data)
 			oc := Cache{DB: db}
 
 			err := oc.Put(c.key, c.value)
 			assertErrClass(t, c.expectedErrors[_redis], err)
 
 			v, err := db.Get([]byte(c.key))
 			assert.NoError(t, err)
-			na := &overlay.Node{}
 
+			na := &overlay.Node{}
 			assert.NoError(t, proto.Unmarshal(v, na))
+
+			{ // see https://github.com/vitessio/vitess/issues/3566
+				na.XXX_sizecache = c.value.XXX_sizecache
+				na.Address.XXX_sizecache = c.value.Address.XXX_sizecache
```

Comment:
also discussed in that issue, or at least something linked from it, was that we should be using proto.Equal instead. Maybe this should be `assert.True(t, proto.Equal(na, &c.value))`?

Reply:
Yes, proto.Equal is better. Somehow I misread the issue, that a newer protobuf version doesn't have that problem, but seems I was wrong. Will fix.